### ISSUE TITLE
panic: runtime error: index out of range in CSS lexer

### ISSUE THREAD

**Issue by lpar** [tdewolff/minify#161]
```
% minify -o public/ -r public
panic: runtime error: index out of range

goroutine 1108 [running]:
github.com/tdewolff/buffer.(*Lexer).Peek(...)
	/Users/meta/Programming/Go/src/github.com/tdewolff/buffer/lexer.go:97
github.com/tdewolff/parse/css.(*Lexer).consumeString(0xc42037a0c0, 0x1120000000001)
	/Users/meta/Programming/Go/src/github.com/tdewolff/parse/css/lex.go:624 +0x1a5
github.com/tdewolff/parse/css.(*Lexer).Next(0xc42037a0c0, 0x114c426, 0xc420387a8b, 0x148a1a1, 0xc4202e8689)
	/Users/meta/Programming/Go/src/github.com/tdewolff/parse/css/lex.go:184 +0x2f8
github.com/tdewolff/parse/css.(*Parser).popToken(0xc4200fa080, 0xc400000001, 0xc4203ee740, 0xc4202e867c, 0xb, 0xc4203ee740)
	/Users/meta/Programming/Go/src/github.com/tdewolff/parse/css/parse.go:142 +0x36
github.com/tdewolff/parse/css.(*Parser).Next(0xc4200fa080, 0xc4202e867c, 0xb, 0x1784, 0xc4203ee740)
	/Users/meta/Programming/Go/src/github.com/tdewolff/parse/css/parse.go:129 +0x10d
github.com/tdewolff/minify/css.(*cssMinifier).minifyGrammar(0xc4200490d8, 0xc4203ee700, 0x1326b01)
	/Users/meta/Programming/Go/src/github.com/tdewolff/minify/css/css.go:67 +0x4a
github.com/tdewolff/minify/css.(*Minifier).Minify(0xc4200168f8, 0xc420215410, 0x148bdc0, 0xc4203ee6c0, 0x148bd80, 0xc4203ee700, 0xc4203fe7b0, 0xc420049190, 0x1284bb1)
	/Users/meta/Programming/Go/src/github.com/tdewolff/minify/css/css.go:58 +0xc9
github.com/tdewolff/minify.(*M).MinifyMimetype(0xc420215410, 0x148a328, 0x8, 0x8, 0x148bdc0, 0xc4203ee6c0, 0x148bd80, 0xc4203ee700, 0xc4203fe7b0, 0x179e, ...)
	/Users/meta/Programming/Go/src/github.com/tdewolff/minify/minify.go:139 +0x223
github.com/tdewolff/minify/html.(*Minifier).Minify(0xc420016900, 0xc420215410, 0x148c080, 0xc4203ee6a0, 0x148c040, 0xc4203ee660, 0x0, 0xc42003ca68, 0x1045e01)
	/Users/meta/Programming/Go/src/github.com/tdewolff/minify/html/html.go:397 +0x1ca7
github.com/tdewolff/minify.(*M).MinifyMimetype(0xc420215410, 0xc42020e2f0, 0x9, 0x10, 0x148c080, 0xc4203ee6a0, 0x148c040, 0xc4203ee660, 0x0, 0xc4185775e8, ...)
	/Users/meta/Programming/Go/src/github.com/tdewolff/minify/minify.go:139 +0x223
github.com/tdewolff/minify.(*M).Minify(0xc420215410, 0x1327c79, 0x9, 0x148c080, 0xc4203ee6a0, 0x148c040, 0xc4203ee660, 0x0, 0x0)
	/Users/meta/Programming/Go/src/github.com/tdewolff/minify/minify.go:130 +0xce
main.minify(0x1327c79, 0x9, 0xc420427100, 0x1, 0x1, 0xc420016930, 0x7, 0xc42040dfb0, 0x2c, 0x0)
	/Users/meta/Programming/Go/src/github.com/tdewolff/minify/cmd/minify/main.go:597 +0xaf7
main.main.func2(0xc420242120, 0xc42003ece0, 0xc420016928, 0xc420427100, 0x1, 0x1, 0xc420016930, 0x7, 0xc42040dfb0, 0x2c)
	/Users/meta/Programming/Go/src/github.com/tdewolff/minify/cmd/minify/main.go:217 +0x8b
created by main.main
	/Users/meta/Programming/Go/src/github.com/tdewolff/minify/cmd/minify/main.go:213 +0x1fd2
```

I just did 
```
go get -u github.com/tdewolff/minify
go get -u github.com/tdewolff/minify/cmd/minify
go install github.com/tdewolff/minify/cmd/minify
```
so version should be `HEAD`.

**Comment by tdewolff** [comment]
This is a regression, could you post the file that causes the problem? That would help me a lot in finding the bug.

**Comment by lpar** [comment]
Is there any way to find out what file that is? It doesn't seem to happen if I call minify on the only CSS file under `public/`.

**Comment by tdewolff** [comment]
It should be a HTML file containing CSS

**Comment by lpar** [comment]
Yes, unfortunately I have 2,818 of those, it's a Hugo site.

**Comment by lpar** [comment]
Oh, you're in luck! It's the top level `index.html`. Here's a Gist: https://gist.github.com/lpar/afcb2bba1336f8292daadeb647e5d765

**Comment by tdewolff** [comment]
This should be fixed now with ac3065f, can you confirm?

**Comment by lpar** [comment]
Yes, fixed thanks! And congratulations on the speed increase in the new version, I can now minify my entire site in almost no time even on a 4 years old laptop.
